cli/compiler.rs

```diff
@@ -218,6 +223,7 @@ pub fn compile_async(
           let res_data = res["data"].as_object().expect(
             "Error decoding compiler response: expected object field 'data'",
           );
+          drop(compiling_job);
```

Comment:
What's this drop thing for? Wouldn't it be dropped otherwise?

Reply:
I'd suggest to to change it to `compiling_job.done()` - although I get why @ry implemented `Drop` trait it's misleading that there's explicit `drop(compiling_job)` call here, also leaving it to be dropped implicit will make it harder to reason about.

Reply:
I've added a comment.